docs/metrics-data-model.md

```diff
@@ -0,0 +1,177 @@
+
+#### Open questions:
+
+* Is the same structure by TagSet required? It could be useful in the case we need to compare a pair of TagSet, where we can just execute single access by hash instead of N comparisons.
+* Should it be possible for any reason to fetch raw data points (not aggregated)? Like `samples := tsdb.GetSamplesBySeries(seriesID, timeStart, timeEnd)`? 
+
+## Collect
+
+#### Open questions:
+
+* Should we change something in `metrics.PushIfNotDone()`?
+* Do we require to identify the time series before invoking `metrics.PushIfNotDone`? So, does the sample require to reference the time series?
+
+## metrics.Ingester
+
+TBD
+
+## Outputs
+
+Outputs should interact with the `tsdb` for fetching a time series snapshot. `series := tsdb.GetSeries()`.
+
+#### Open questions:
+
+* Should we query by timeframe so we can filter out time series without samples and avoid pushing the time series without a changed value?
+
+#### Aggregation
+
+Using time series from the outputs as described is already a form of aggregation. 
+
+#### Open questions:
+
+* Do we require high accurate timeframe for aggregation? Currently, the aggregation window is defined by the `PeriodicFlusher` but in the high concurrent world like `k6` it could be not exactly the expected timeframe (e.g. instead to be now-1s <= t < now).
+
+#### Concurrency
+
+In the case, that the write endpoint for flushing data is slow and it returns the response in a greater interval than the interval set for the `PeriodicFlusher` then we could push a not optimal aggregated window where the aggregation is based on a wider range.
+
+Is this an issue? If yes:
+
+* In the case the Prometheus instance slows down in a limited range then we could mitigate it by adding concurrent flush. It requires adding a label per goroutine (e.g. a Worker ID ?) so we can avoid hitting out-of-order issues. 
+
+#### Prometheus
+
+##### Trend
+
+[:bulb:] We could potentially mitigate the sparse histogram by making the histogram configurable by adding two options:
+
+* UseTrendAsHistogram: where it says to the output to use the histogram for Trend.
+* HistogramBuckets: it would require a map between Trend metrics' name and the expected histogram.
+
+In the case, we couldn't directly implement the right dynamic histogram, this would help advanced users that know how the SUT behaves to use histogram instead of the 6 gauges workaround.
+
+##### End of series markers
+
+Is the marker required and/or useful for the output?
+
+#### Thresholds
+
+Thresholds should act as an output, it should work on top of a time series data layer. They get the latest value for the time series and check if the value triggers the threshold based on the defined condition.
+
+## Known issues
+
+* Name and URL tags: `k6` is tagging HTTP requests with the URL. It will create a high cardinality issue for the time series data model.
```

Comment:
@na-- or even better, when we will have the metadata concept then we can just move it there

Reply:
Yes, that's what I'd like, though there will be some... complications :disappointed: The [`iter`, `vu` (and maybe `ip`) tags](https://github.com/grafana/k6/blob/556747fbe585c981507904d634e705c68891e663/metrics/system_tag.go#L103-L106) should probably also be metadata, not tags, but it's a bit too early to see how all of this will work out.